src/googmodule.ts

```diff
@@ -197,6 +197,37 @@ function rewriteModuleExportsAssignment(expr: ts.ExpressionStatement) {
       expr);
 }
 
+/**
+ * Convert a series of comma-separated expressions
+ *   x = foo, y(), z.bar();
+ * with statements
+ *   x = foo; y(); z.bar();
+ * This is for handling in particular the case where
+ *   exports.x = ..., exports.y = ...;
+ * which Closure rejects.
+ *
+ * @return An array of statements if it converted, or null otherwise.
+ */
+function rewriteCommaExpressions(expr: ts.ExpressionStatement): ts.Statement[]|null {
+  // Early exit if the outer statement isn't a comma statement.
+  if (!ts.isBinaryExpression(expr.expression)) return null;
+  if (expr.expression.operatorToken.kind !== ts.SyntaxKind.CommaToken) return null;
+
+  const stmts: ts.Statement[] = [];
+  visit(expr.expression);
+  function visit(expr: ts.Expression) {
```

Comment:
the usual pattern here is to have `visit()` return a `ts.Node|ts.Node[]`, and calls to `visit` use its return value recursively. It took me a bit to understand that you avoid this to flatten the comma expression by explicit pushes on stmts. Maybe you could add a comment explaining that and why you're deviating from the normal pattern?

Reply:
I'm not sure I followed you here, can you point me at one of the examples of the normal pattern?

Reply:
Here's an example:
https://github.com/angular/tsickle/blob/master/src/jsdoc_transformer.ts#L932

Note how `visitor` returns `ts.Node|ts.Node[]`, and (with more precise types) so do the functions it dispatches to. The functions then always return a node or list of nodes, either the input unchanged, or the modified nodes.

Reply:
PTAL